File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/MobileQuickStart/samples/WritingControls/vb/AcmeMenu.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------

Imports System
Imports System.Web.UI
Imports System.Web.UI.MobileControls

Namespace Acme.CustomControlsVB

    Public Class Menu : Inherits System.Web.UI.MobileControls.List
    
        Overrides Protected Sub OnLoad(e As EventArgs)

            If (Not Page.IsPostBack)
	    
	        MyBase.OnLoad(e)
	        Dim thisForm As Form = Form
	        Dim testForm As Form
	        For Each testForm in MobilePage.Forms
	     	 
		    If (Not(testForm Is thisForm))
		     
		        Dim item As MobileListItem = New MobileListItem() 
		        Dim formTitle As String = testForm.Title
		        If (formTitle.Length > 0)
			    item.Text = formTitle
		        Else
			    item.Text = testForm.ID 
		        End If
		        item.Value = testForm.UniqueID
		        Items.Add(item)
		    
		    End If
		    
	        Next

            End If

            AddHandler Me.ItemCommand, AddressOf Me.HandleCommand
        
        End Sub

        Protected Sub HandleCommand(sender As Object, e As ListCommandEventArgs)
        
            Dim ctl As Form = CType(MobilePage.FindControl(e.ListItem.Value), Form)
            If (Not (ctl Is Nothing))
                MobilePage.ActiveForm = ctl
            End If
            
        End Sub
        
    End Class
    
End Namespace

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/MobileQuickStart/samples/WritingControls/js/AcmeMenu.js
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

import System;
import System.Web.UI.MobileControls;
package Acme.CustomControlsJS
{
    public class Menu extends System.Web.UI.MobileControls.List
    {
        protected override function OnLoad( e:EventArgs)
        {
            if (!Page.IsPostBack)
            {
                super.OnLoad(e);
                var thisForm : System.Web.UI.MobileControls.Form = this.Form;
                for (var form : System.Web.UI.MobileControls.Form in MobilePage.Forms)
                {
                    if (form != thisForm)
                    {
                        var item : MobileListItem = new MobileListItem();
                        var formTitle : String = form.Title;
                        if (formTitle.Length > 0)
                        {
                            item.Text = formTitle;
                        }
                        else
                        {
                            item.Text = form.ID;
                        }
                        item.Value = form.UniqueID;
                        Items.Add(item);
                    }
                }
            }
			this.add_ItemCommand(this.HandleCommand);
        }

        protected function HandleCommand(sender:Object , e:ListCommandEventArgs )
        {
            var form : System.Web.UI.MobileControls.Form = MobilePage.FindControl(e.ListItem.Value); 
            if (form != null)
            {
                MobilePage.ActiveForm = form;
            }
        }
    }
}
